### ISSUE TITLE
Extension possibly not notifying Gnome that the desktop's workspace has changed when hiding the top bar

### ISSUE THREAD

**Issue by necropolina** [mlutfy/hidetopbar#341]
https://github.com/pop-os/shell/issues/1113

According to this issue, Pop Shell's window tiling should make windows dynamically fill and retreat from the space that the top bar occupies when it gets enabled/disabled. This does work when disabling the extension - windows will shorten so as to give space for the top bar. But, when re-enabling extension (or logging back in after the screen gets locked), a gap is left at the top of the screen where the top bar was.

**Comment by necropolina** [comment]
Actually, looks like it might be a pop shell issue, now that I look at it. Will leave this open for you to see tho, just in case.

**Comment by jacksongoode** [comment]
This happens with select applications (like VSCode) in my experience (Fedora 35).

**Comment by amfern** [comment]
Actually this happens to me all the time, after login from lock screen there is a gap the size top bar.
gnome 41.

**Comment by AndreasPB** [comment]
Yep, same. Every time I unlock my screen I have to do a window-shuffling-routine to get the tiling manager to understand that the topbar space is free

**Comment by amfern** [comment]
But i see the gap without pop-shell, just vanilla gnome 41, with extension that hides the top bar like hidetopbar or justperfection
